test/tests/functional/pbs_cray_suspend_resume.py

```diff
@@ -93,3 +96,373 @@ def test_exclusive_job_not_suspended(self):
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=j1id)
+
+        j2 = Job(TEST_USER, {ATTR_l + '.select': '%d:ncpus=1' % nv,
+                             ATTR_l + '.place': 'scatter',
+                             ATTR_l + '.walltime': 3600,
+                             ATTR_q: 'expressq'})
+        j2id = self.server.submit(j2)
+        self.server.expect(JOB, {ATTR_state: 'S'}, id=j1id)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=j2id)
+
+        j3 = Job(TEST_USER, {ATTR_l + '.select': '%d:ncpus=1' % nv,
+                             ATTR_l + '.place': 'scatter',
+                             ATTR_l + '.walltime': 3600,
+                             ATTR_q: 'expressq2'})
+        j3id = self.server.submit(j3)
+        self.server.expect(JOB, {ATTR_state: 'S'}, id=j2id)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=j3id)
+
+        j4 = Job(TEST_USER, {ATTR_l + '.select': '%d:ncpus=1' % nv,
+                             ATTR_l + '.place': 'scatter',
+                             ATTR_l + '.walltime': 3600,
+                             ATTR_q: 'expressq3'})
+        j4id = self.server.submit(j4)
+        self.server.expect(JOB, {ATTR_state: 'S'}, id=j3id)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=j4id)
+
+        self.server.delete(j4id)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=j3id)
+
+    def test_job_accounted_limit_usage(self):
+        """
+        Test that a job is correctly accounted limit usage
+        """
+        try:
+            self.server.status(QUEUE, id='expressq', logerr=False)
+            self.server.manager(MGR_CMD_DELETE, QUEUE, None, 'expressq')
+        except (PbsStatusError, PbsManagerError):
+            pass
+
+        a = {'queue_type': 'execution',
+             'priority': 150,
+             'enabled': 'True',
+             'started': 'True'}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, 'expressq')
+
+        a = {'max_run_res_soft.ncpus': '[p:PBS_GENERIC=1]',
+             'eligible_time_enable': 1}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+        # Count the compute nodes
+        nv = self.num_compute_vnodes()
+
+        j1 = Job(TEST_USER, {ATTR_l + '.select': '%d:ncpus=1' % nv,
+                             ATTR_l + '.place': 'scatter',
+                             ATTR_l + '.walltime': '3600',
+                             ATTR_project: 'A'})
+        j1id = self.server.submit(j1)
+
+        # Ensure j2 is started after j1 due to preempt_sort:
+        # min_time_since_start
+        time.sleep(1)
+        d = self.server.expect(JOB, {ATTR_state: 'R'}, id=j1id)
+        self.assertTrue(d)
+
+        j2 = Job(TEST_USER, {ATTR_l + '.select': '%d:ncpus=1' % nv,
+                             ATTR_l + '.place': 'scatter',
+                             ATTR_l + '.walltime': '3600',
+                             ATTR_project: 'A'})
+        j2id = self.server.submit(j2)
+        d = self.server.expect(JOB, {ATTR_state: 'R'}, id=j2id)
```

Comment:
This will not work, If J1 is taking up all the compute nodes then J2 can not run unless it is submitted into high priority job and I don't think you intend to do that here.

Reply:
I have taken out that test which is redundant because it is checking suspension by preemption which is already being done in test_multi_express.